AdfDataflowFilePrettifier/Prettifier.cs

```diff
@@ -78,36 +75,96 @@ private static (string, string, string) SplitFileIntoInterestingSections(string
                 );
         }
 
-
-        private const char InnerMagicIndicatorCharacter = '\x05'; // We want to be able to distinguish our 'created' \n characters, etc., from the 'real' ones, so that we can reverse the process later.
-        private const char OuterMagicIndicatorCharacter = '\x06'; // And we also want to do this slightly different at the edges of the script.
-        private const char CRMagicIndicatorCharacter = '\x07';    // Due to limitations of kDiff we want to flag CR (\r) characters separately too.
+        private const string BlockStartString = "\n/// PRETTIFIED SCRIPT START MARKER /// DO NOT ALTER THIS LINE ///\n";
+        private const string BlockEndString = "\n/// PRETTIFIED SCRIPT END MARKER /// DO NOT ALTER THIS LINE ///\n";
 
         private static string PrettifyDataFlowScriptDefinition(string scriptLineToPrettify)
         {
-            var prettierLine = scriptLineToPrettify;
-
-            var markedNewLine = OuterMagicIndicatorCharacter + "\n";
-            prettierLine = prettierLine.Replace("\\r\\n", CRMagicIndicatorCharacter + "\n");

```

Comment:
Yes, you have.

Hmm ... so ... this is a specific thing to support `kdiff` because `kdiff` is a bit bad at line endings.
You can pick your resultant line endings, but you can't end up with mixed LEs. :(
So these `"\\r\\n"` strings get turned into `"\r\n"` by the prettifier, and then turned into `"\n"` by kdiff, and then turned into `"\\n"` by the uglifier. But `"\\r\\n"` != `"\\n"`. [citation needed]

I should have created a test for that case :(

I do still need that code ... but it shouldn't be hidden away pretending to be "part of core prettification/uglification". It should be an explicit step in the process, layered separately.

Do you have any suggestions *other* than adding some sort of magic indicator (ASCII control character or otherwise)?

Reply:
Ouch. This seems like a kdiff issue! But actually, more generally, isn't there an issue with kdiff trampling the line endings and if you make any manual changes in kdiff? Like resolving any merges manually?

Azure (I would hope) would only ever use one line ending anyway, so wouldn't the best thing to do be to just be sneaky and replace all line endings with the azure ones when we uglify? Or alternatively boot up kdiff with a \r\n line ending style http://kdiff3.sourceforge.net/doc/options.html using the cs command line flag http://kdiff3.sourceforge.net/doc/documentation.html ? 

Also - we should imo tell git attributes to turn autocrlf off for these json files if it's not already, in case that impacts this process at all? (although existing checkouts with autocrlf on might have issues).

Reply:
There's also an argument that we don't need the output of prettifying and uglifying to neccessarily be opposites - the process can simply normalize the output, as long as it's still valid ADF code?

Ie the mergetool could also clean up line endings and that would be okay?